python/cudf/cudf/core/column/column.py

```diff
@@ -880,6 +880,11 @@ def distinct_count(self, method="sort", dropna=True):
     def astype(self, dtype, **kwargs):
         if is_categorical_dtype(dtype):
             return self.as_categorical_column(dtype, **kwargs)
+        elif is_list_dtype(dtype):
+            if not is_list_dtype(self.dtype):
+                raise ValueError(f"Cannot cast list column to {dtype}")
+            else:
+                return self
```

Comment:
Yeah good catch. Is that basically the only kind of casting we should can/support for list columns, i.e., casting the leaf values?

Reply:
Only other semi-normal thing I can think of would be being able to do `String` <--> `List(int8)` including at various levels of nesting.

Otherwise in a bit more niche area, you could imagine someone wanting to `view` as opposed to `cast` from say `int32` --> `int64` assuming there's a divisible number of elements in each List.